src/authorizers/lambda.ts

```diff
@@ -48,6 +52,28 @@ export class LambdaAuthorizer extends Construct {
       }()),
     };
   }
-}
 
+  public grantFunctionInvoke(api: IRestApi): void {
+    const fn = Function.fromFunctionAttributes(this, `ImportFunction${this.id}`,
+      {
+        functionArn: this.fn.functionArn,
+        sameEnvironment: true,
+      });
 
+    const authorizerArn = Stack.of(this).formatArn({
+      service: 'execute-api',
+      resource: api.restApiId,
+      resourceName: 'authorizers/*',
+    });
+
+    fn.grantInvoke(new ServicePrincipal('apigateway.amazonaws.com',
+      {
+        conditions: {
+          ArnLike: {
+            'aws:SourceArn': authorizerArn,
+          },
+        },
+      },
+    ));
+  }
```

Comment:
Is the `Function.fromFunctionAttributes` necessary? Could we just use the `grantInvoke` method [of `IFunction` interface](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.IFunction.html#grantwbrinvokeidentity) via the `this.fn` reference?

Reply:
Yeah, this is exactly the hacky way to avoid circular dependencies.